src/qibocal/auto/task.py

```diff
@@ -134,6 +135,18 @@ def run(
         completed = Completed(self, Normal(), folder)
         task_qubits = self._allocate_local_qubits(qubits, platform)
 
+        # get parent and child class data_members
+        parameters_keys = set(
+            self.operation.parameters_type.__dict__["__annotations__"].keys()
+        ) | set(Parameters.__dict__["__annotations__"].keys())
+        # read all settings defined in platform-specific qibolab runcard
+        for setting, value in platform.settings.__dict__.items():
+            # save values non defined in qibocal runcard,
+            # but that are used by the class
+            if setting not in self.action.parameters:
+                if setting in parameters_keys:
+                    self.action.parameters[setting] = value
```

Comment:
I think that this can be done more easily...
What about this?
```suggestion
        if self.parameters.nshots is None:
            self.parameters.nshots = platform.settings.nshots

        if self.parameters.relaxation_time is None:
            self.parameters.relaxation_time = platform.settings.relaxation_time
```
I tested and it seems to work as expected.
You can also remove the deepcopy that you did before I believe.

Reply:
My only concern with this solution is that if in the future we add more parameters to `qibolab` runcards, it is going to pass under the radar, since this behaviour is not well tested. 
By the way, I’m not satisfied with the tests I drafted yesterday, so I need to think some way to implement some testing.

Reply:
Currently I believe that the only common parameters between all the routines are `nshots` and `relaxation_time`.

Reply:
For testing, we just need to add some routines in the test runcard settings those fields to none, `rabi_amplitude` would be one that could work as it uses `params.nshots` for the fit.